doc/developer/60-debugging-and-testing.md

```diff
@@ -20,6 +20,33 @@ Example mounting with `Fuse3` on mount point `/mnt/test`
 > **_NOTE_** &nbsp;
 > The `uid` and `gid` are the `user id` and `group id` of the `cvmfs` user.
 > They will be different on every system but the here provided code discovers them automatically.
+> 
+> A debug version of the client is automatically built. No special build parameters are needed.
+
+### Server
+Live debugging of the server is more complicated. For server and server unit tests a special debug build must be created using `BUILD_SERVER_DEBUG=ON`.
+
+To run the server in debug mode, the environment variable `CVMFS_SERVER_DEBUG` must be set. It allows switching between 3 different debug modes 
+- `CVMFS_SERVER_DEBUG=1`
+  - In case something breaks, provide a GDB prompt
+  - Underlying command: `gdb --quiet --eval-command=run --eval-command=quit --args cvmfs_swissknife_debug`
+- `CVMFS_SERVER_DEBUG=2`
+  - Attach GDB and provide a prompt WITHOUT actual running the program (the user starts the run, allows setting breakpoints first)
+  - Underlying command: `gdb --quiet --args cvmfs_swissknife_debug`
+- `CVMFS_SERVER_DEBUG=3`
+  - Run debug version of the server (without GDB)
+  - Underlying command: `cvmfs_swissknife_debug`
+
+For serious debugging `CVMFS_SERVER_DEBUG=2` is the preferred choice. `CVMFS_SERVER_DEBUG=1` only opens GDB prompt when exiting the program but in many cases `cvmfs` has proper error handling, exiting gracefully and as such finding the code section responsible for the error is not obvious.
```

Comment:
In a first pass, `CVMFS_SERVER_DEBUG=3` is quite helpful because it prints all the debug messages that are normally surpressed.